### PR TITLE
Mitigate the impact of pgcopy timeout

### LINKED ISSUE #2088: Importer timed out during pgcopy
**Detailed Description**

Importer timed out during pgcopy `transaction` table with an error after 7.5 minutes in performance environment. The actual stacktrace seems to be a race condition between server timeout and calling `cancelCopy` in a finally block. The real problem seems to be that pgcopy froze for 7.5 minutes while not doing anything as far as metrics can tell. There is an `automatic analyze of table "mirror_node.public.token_transfer"` around the time the pgcopy starts, but that might not be related.

Investigation reveals that while the stacktrace seems rare the actual pgcopy timeout issue has occurred a number of times as far back as a month ago (the farthest we have logs for):

screen_shot_2021-06-04_at_1 59 55_pm

**Actual Behavior**
```
Jun 04 12:33:32 mirrornode-performance-parser-1 java[28770]: 2021-06-04T06:33:32.045-0600 ERROR scheduling-6 c.h.m.i.p.r.RecordFileParser Error parsing file 2021-06-04T12_25_44.013647000Z.rcd com.hedera.mirror.importer.exception.ParserException: Error copying 19148 items to table transaction\
        at com.hedera.mirror.importer.parser.PgCopy.copy(PgCopy.java:99)\
        at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.executeBatches(SqlEntityListener.java:192)\
        at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.onEnd(SqlEntityListener.java:160)\
        at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.onEnd(SqlEntityListener.java:69)\
        at com.hedera.mirror.importer.parser.record.RecordFileParser.parse(RecordFileParser.java:165)\
        at com.hedera.mirror.importer.parser.record.RecordFileParser$$FastClassBySpringCGLIB$$8b9310db.invoke(<generated>)\
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)\
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:779)\
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)\
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)\
        at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)\
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388)\
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)\
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)\
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)\
        at org.springframework.retry.interceptor.RetryOperationsInterceptor$1.doWithRetry(RetryOperationsInterceptor.java:93)\
        at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:329)\
        at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:209)\
        at org.springframework.retry.interceptor.RetryOperationsInterceptor.invoke(RetryOperationsInterceptor.java:119)\
        at org.springframework.retry.annotation.AnnotationAwareRetryOperationsInterceptor.invoke(AnnotationAwareRetryOperationsInterceptor.java:163)\
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)\
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)\
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:692)\
        at com.hedera.mirror.importer.parser.record.RecordFileParser$$EnhancerBySpringCGLIB$$4c111cfe.parse(<generated>)\
        at jdk.internal.reflect.GeneratedMethodAccessor155.invoke(Unknown Source)\
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)\
        at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:171)\
        at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.invoke(InvocableHandlerMethod.java:120)\
        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper$HandlerMethod.invoke(MessagingMethodInvokerHelper.java:1105)\
        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper.invokeHandlerMethod(MessagingMethodInvokerHelper.java:583)\
        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper.processInternal(MessagingMethodInvokerHelper.java:478)\
        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper.process(MessagingMethodInvokerHelper.java:356)\
        at org.springframework.integration.handler.MethodInvokingMessageProcessor.processMessage(MethodInvokingMessageProcessor.java:108)\
        at org.springframework.integration.handler.ServiceActivatingHandler.handleRequestMessage(ServiceActivatingHandler.java:105)\
        at org.springframework.integration.handler.AbstractReplyProducingMessageHandler.handleMessageInternal(AbstractReplyProducingMessageHandler.java:134)\
        at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:56)\
        at org.springframework.integration.endpoint.PollingConsumer.handleMessage(PollingConsumer.java:158)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.messageReceived(AbstractPollingEndpoint.java:449)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.doPoll(AbstractPollingEndpoint.java:435)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.pollForMessage(AbstractPollingEndpoint.java:387)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$null$4(AbstractPollingEndpoint.java:331)\
        at org.springframework.integration.util.ErrorHandlingTaskExecutor.lambda$execute$0(ErrorHandlingTaskExecutor.java:57)\
        at org.springframework.core.task.SyncTaskExecutor.execute(SyncTaskExecutor.java:50)\
        at org.springframework.integration.util.ErrorHandlingTaskExecutor.execute(ErrorHandlingTaskExecutor.java:55)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$createPoller$5(AbstractPollingEndpoint.java:328)\
        at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)\
        at org.springframework.scheduling.concurrent.ReschedulingRunnable.run(ReschedulingRunnable.java:95)\
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)\
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)\
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)\
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\
        at java.base/java.lang.Thread.run(Thread.java:829)\
Caused by: org.postgresql.util.PSQLException: Database connection failed when canceling copy operation\
        at org.postgresql.core.v3.QueryExecutorImpl.cancelCopy(QueryExecutorImpl.java:971)\
        at org.postgresql.core.v3.CopyOperationImpl.cancelCopy(CopyOperationImpl.java:36)\
        at com.hedera.mirror.importer.parser.PgCopy.copy(PgCopy.java:92)\
        ... 53 more\
Caused by: java.net.SocketException: Broken pipe (Write failed)\
        at java.base/java.net.SocketOutputStream.socketWrite0(Native Method)\
        at java.base/java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:110)\
        at java.base/java.net.SocketOutputStream.write(SocketOutputStream.java:150)\
        at java.base/sun.security.ssl.SSLSocketOutputRecord.deliver(SSLSocketOutputRecord.java:319)\
        at java.base/sun.security.ssl.SSLSocketImpl$AppOutputStream.write(SSLSocketImpl.java:1212)\
        at java.base/java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:81)\
        at java.base/java.io.BufferedOutputStream.flush(BufferedOutputStream.java:142)\
        at org.postgresql.core.PGStream.flush(PGStream.java:665)\
        at org.postgresql.core.v3.QueryExecutorImpl.cancelCopy(QueryExecutorImpl.java:948)\
        ... 55 more
Jun 04 12:33:32 mirrornode-performance-parser-1 java[28770]: 2021-06-04T06:33:32.045-0600 INFO scheduling-6 c.h.m.i.p.r.RecordFileParser Finished parsing 19148 transactions from record file 2021-06-04T12_25_44.013647000Z.rcd in 7.594 min. Success: false
Jun 04 12:33:32 mirrornode-performance-parser-1 java[28770]: 2021-06-04T06:33:32.055-0600 WARN scheduling-6 c.z.h.p.ProxyConnection mirror_node - Connection org.postgresql.jdbc.PgConnection@6f4fa8f5 marked as broken because of SQLSTATE(08006), ErrorCode(0) org.postgresql.util.PSQLException: An I/O error occurred while sending to the backend.\
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:350)\
        at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:849)\
        at org.postgresql.jdbc.PgConnection.rollback(PgConnection.java:892)\
        at com.zaxxer.hikari.pool.ProxyConnection.rollback(ProxyConnection.java:396)\
        at com.zaxxer.hikari.pool.HikariProxyConnection.rollback(HikariProxyConnection.java)\
        at org.hibernate.resource.jdbc.internal.AbstractLogicalConnectionImplementor.rollback(AbstractLogicalConnectionImplementor.java:121)\
        at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl$TransactionDriverControlImpl.rollback(JdbcResourceLocalTransactionCoordinatorImpl.java:304)\
        at org.hibernate.engine.transaction.internal.TransactionImpl.rollback(TransactionImpl.java:142)\
        at org.springframework.orm.jpa.JpaTransactionManager.doRollback(JpaTransactionManager.java:589)\
        at org.springframework.transaction.support.AbstractPlatformTransactionManager.processRollback(AbstractPlatformTransactionManager.java:835)\
        at org.springframework.transaction.support.AbstractPlatformTransactionManager.rollback(AbstractPlatformTransactionManager.java:809)\
        at org.springframework.transaction.interceptor.TransactionAspectSupport.completeTransactionAfterThrowing(TransactionAspectSupport.java:672)\
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:392)\
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)\
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)\
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)\
        at org.springframework.retry.interceptor.RetryOperationsInterceptor$1.doWithRetry(RetryOperationsInterceptor.java:93)\
        at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:329)\
        at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:209)\
        at org.springframework.retry.interceptor.RetryOperationsInterceptor.invoke(RetryOperationsInterceptor.java:119)\
        at org.springframework.retry.annotation.AnnotationAwareRetryOperationsInterceptor.invoke(AnnotationAwareRetryOperationsInterceptor.java:163)\
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)\
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)\
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:692)\
        at com.hedera.mirror.importer.parser.record.RecordFileParser$$EnhancerBySpringCGLIB$$4c111cfe.parse(<generated>)\
        at jdk.internal.reflect.GeneratedMethodAccessor155.invoke(Unknown Source)\
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)\
        at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:171)\
        at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.invoke(InvocableHandlerMethod.java:120)\
        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper$HandlerMethod.invoke(MessagingMethodInvokerHelper.java:1105)\
        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper.invokeHandlerMethod(MessagingMethodInvokerHelper.java:583)\
        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper.processInternal(MessagingMethodInvokerHelper.java:478)\
        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper.process(MessagingMethodInvokerHelper.java:356)\
        at org.springframework.integration.handler.MethodInvokingMessageProcessor.processMessage(MethodInvokingMessageProcessor.java:108)\
        at org.springframework.integration.handler.ServiceActivatingHandler.handleRequestMessage(ServiceActivatingHandler.java:105)\
        at org.springframework.integration.handler.AbstractReplyProducingMessageHandler.handleMessageInternal(AbstractReplyProducingMessageHandler.java:134)\
        at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:56)\
        at org.springframework.integration.endpoint.PollingConsumer.handleMessage(PollingConsumer.java:158)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.messageReceived(AbstractPollingEndpoint.java:449)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.doPoll(AbstractPollingEndpoint.java:435)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.pollForMessage(AbstractPollingEndpoint.java:387)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$null$4(AbstractPollingEndpoint.java:331)\
        at org.springframework.integration.util.ErrorHandlingTaskExecutor.lambda$execute$0(ErrorHandlingTaskExecutor.java:57)\
        at org.springframework.core.task.SyncTaskExecutor.execute(SyncTaskExecutor.java:50)\
        at org.springframework.integration.util.ErrorHandlingTaskExecutor.execute(ErrorHandlingTaskExecutor.java:55)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$createPoller$5(AbstractPollingEndpoint.java:328)\
        at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)\
        at org.springframework.scheduling.concurrent.ReschedulingRunnable.run(ReschedulingRunnable.java:95)\
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)\
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)\
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)\
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\
        at java.base/java.lang.Thread.run(Thread.java:829)\
Caused by: java.net.SocketException: Broken pipe (Write failed)\
        at java.base/java.net.SocketOutputStream.socketWrite0(Native Method)\
        at java.base/java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:110)\
        at java.base/java.net.SocketOutputStream.write(SocketOutputStream.java:150)\
        at java.base/sun.security.ssl.SSLSocketOutputRecord.deliver(SSLSocketOutputRecord.java:319)\
        at java.base/sun.security.ssl.SSLSocketImpl$AppOutputStream.write(SSLSocketImpl.java:1212)\
        at java.base/java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:81)\
        at java.base/java.io.BufferedOutputStream.flush(BufferedOutputStream.java:142)\
        at org.postgresql.core.PGStream.flush(PGStream.java:665)\
        at org.postgresql.core.v3.QueryExecutorImpl.sendSync(QueryExecutorImpl.java:1470)\
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:321)\
        ... 54 more
        Jun 04 12:33:32 mirrornode-performance-parser-1 java[28770]: 2021-06-04T06:33:32.062-0600 ERROR scheduling-6 o.s.t.i.TransactionInterceptor Application exception overridden by rollback exception com.hedera.mirror.importer.exception.ParserException: Error copying 19148 items to table transaction\
        at com.hedera.mirror.importer.parser.PgCopy.copy(PgCopy.java:99)\
        at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.executeBatches(SqlEntityListener.java:192)\
        at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.onEnd(SqlEntityListener.java:160)\
        at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.onEnd(SqlEntityListener.java:69)\
        at com.hedera.mirror.importer.parser.record.RecordFileParser.parse(RecordFileParser.java:165)\
        at com.hedera.mirror.importer.parser.record.RecordFileParser$$FastClassBySpringCGLIB$$8b9310db.invoke(<generated>)\
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)\
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:779)\
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)\
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)\
        at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)\
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388)\
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)\
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)\
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)\
        at org.springframework.retry.interceptor.RetryOperationsInterceptor$1.doWithRetry(RetryOperationsInterceptor.java:93)\
        at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:329)\
        at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:209)\
        at org.springframework.retry.interceptor.RetryOperationsInterceptor.invoke(RetryOperationsInterceptor.java:119)\
        at org.springframework.retry.annotation.AnnotationAwareRetryOperationsInterceptor.invoke(AnnotationAwareRetryOperationsInterceptor.java:163)\
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)\
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750)\
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:692)\
        at com.hedera.mirror.importer.parser.record.RecordFileParser$$EnhancerBySpringCGLIB$$4c111cfe.parse(<generated>)\
        at jdk.internal.reflect.GeneratedMethodAccessor155.invoke(Unknown Source)\
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)\
        at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:171)\
        at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.invoke(InvocableHandlerMethod.java:120)\
        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper$HandlerMethod.invoke(MessagingMethodInvokerHelper.java:1105)\
        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper.invokeHandlerMethod(MessagingMethodInvokerHelper.java:583)\
        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper.processInternal(MessagingMethodInvokerHelper.java:478)\
        at org.springframework.integration.handler.support.MessagingMethodInvokerHelper.process(MessagingMethodInvokerHelper.java:356)\
        at org.springframework.integration.handler.MethodInvokingMessageProcessor.processMessage(MethodInvokingMessageProcessor.java:108)\
        at org.springframework.integration.handler.ServiceActivatingHandler.handleRequestMessage(ServiceActivatingHandler.java:105)\
        at org.springframework.integration.handler.AbstractReplyProducingMessageHandler.handleMessageInternal(AbstractReplyProducingMessageHandler.java:134)\
        at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:56)\
        at org.springframework.integration.endpoint.PollingConsumer.handleMessage(PollingConsumer.java:158)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.messageReceived(AbstractPollingEndpoint.java:449)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.doPoll(AbstractPollingEndpoint.java:435)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.pollForMessage(AbstractPollingEndpoint.java:387)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$null$4(AbstractPollingEndpoint.java:331)\
        at org.springframework.integration.util.ErrorHandlingTaskExecutor.lambda$execute$0(ErrorHandlingTaskExecutor.java:57)\
        at org.springframework.core.task.SyncTaskExecutor.execute(SyncTaskExecutor.java:50)\
        at org.springframework.integration.util.ErrorHandlingTaskExecutor.execute(ErrorHandlingTaskExecutor.java:55)\
        at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$createPoller$5(AbstractPollingEndpoint.java:328)\
        at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)\
        at org.springframework.scheduling.concurrent.ReschedulingRunnable.run(ReschedulingRunnable.java:95)\
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)\
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)\
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)\
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\
        at java.base/java.lang.Thread.run(Thread.java:829)\
Caused by: org.postgresql.util.PSQLException: Database connection failed when canceling copy operation\
        at org.postgresql.core.v3.QueryExecutorImpl.cancelCopy(QueryExecutorImpl.java:971)\
        at org.postgresql.core.v3.CopyOperationImpl.cancelCopy(CopyOperationImpl.java:36)\
        at com.hedera.mirror.importer.parser.PgCopy.copy(PgCopy.java:92)\
        ... 53 more\
Caused by: java.net.SocketException: Broken pipe (Write failed)\
        at java.base/java.net.SocketOutputStream.socketWrite0(Native Method)\
        at java.base/java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:110)\
        at java.base/java.net.SocketOutputStream.write(SocketOutputStream.java:150)\
        at java.base/sun.security.ssl.SSLSocketOutputRecord.deliver(SSLSocketOutputRecord.java:319)\
        at java.base/sun.security.ssl.SSLSocketImpl$AppOutputStream.write(SSLSocketImpl.java:1212)\
        at java.base/java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:81)\
        at java.base/java.io.BufferedOutputStream.flush(BufferedOutputStream.java:142)\
        at org.postgresql.core.PGStream.flush(PGStream.java:665)\
        at org.postgresql.core.v3.QueryExecutorImpl.cancelCopy(QueryExecutorImpl.java:948)\
        ... 55 more
            Jun 04 12:33:32 mirrornode-performance-parser-1 java[28770]: 2021-06-04T06:33:32.856-0600 INFO scheduling-6 c.h.m.i.p.PgCopy Copied 19148 rows to transaction table in 313.8 ms
Jun 04 12:33:33 mirrornode-performance-parser-1 java[28770]: 2021-06-04T06:33:33.102-0600 INFO scheduling-6 c.h.m.i.p.PgCopy Copied 57477 rows to crypto_transfer table in 245.6 ms
Jun 04 12:33:33 mirrornode-performance-parser-1 java[28770]: 2021-06-04T06:33:33.321-0600 INFO scheduling-6 c.h.m.i.p.PgCopy Copied 37468 rows to token_transfer table in 219.2 ms
Jun 04 12:33:33 mirrornode-performance-parser-1 java[28770]: 2021-06-04T06:33:33.352-0600 INFO scheduling-6 c.h.m.i.p.r.e.s.SqlEntityListener Inserted 24 entities in 30.64 ms
Jun 04 12:33:33 mirrornode-performance-parser-1 java[28770]: 2021-06-04T06:33:33.352-0600 INFO scheduling-6 c.h.m.i.p.r.e.s.SqlEntityListener Completed batch inserts in 810.0 ms
Jun 04 12:33:33 mirrornode-performance-parser-1 java[28770]: 2021-06-04T06:33:33.354-0600 INFO scheduling-6 c.h.m.i.p.r.RecordFileParser Finished parsing 19148 transactions from record file 2021-06-04T12_25_44.013647000Z.rcd in 1.039 s. Success: true
Jun 04 12:33:33 mirrornode-performance-parser-1 java[28770]: 2021-06-04T06:33:33.804-0600 INFO scheduling-6 c.h.m.i.p.PgCopy Copied 17536 rows to transaction table in 283.8 ms
```

**Expected Behavior**

**Environment:**
- Java: AdoptOpenJDK 11.0.11
- OS: Ubuntu 18.04
- Version: 0.34.0-rc1

**Additional Context**

### PR DESCRIPTION
**Pull request by xin-hedera**
**Detailed description**:
Add timeout to `Transactional` annotation to timeout problematic pgcopy. 

**Which issue(s) this PR fixes**:
Fixes #2088 

**Special notes for your reviewer**:
we should adjust down the account balance parser `transactionTimeout` from 300s when the balance parsing is optimized.

**Checklist**

- [x] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/AbstractParserProperties.java [hunk 1]
```diff
@@ -50,6 +50,8 @@
     @NotNull
     protected RetryProperties retry = new RetryProperties();
 
+    @Min(30)
+    protected int transactionTimeout = 30;
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/AbstractParserProperties.java:54]: We should use Duration like we do everywhere else. Can use `@DurationUnit(SECONDS)` to change default if no unit specified from ms to s.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/balance/AccountBalanceFileParser.java [hunk 2]
```diff
@@ -76,7 +76,7 @@ public AccountBalanceFileParser(MeterRegistry meterRegistry, BalanceParserProper
             maxDelayExpression = "#{@balanceParserProperties.getRetry().getMaxBackoff().toMillis()}",
             multiplierExpression = "#{@balanceParserProperties.getRetry().getMultiplier()}"),
             maxAttemptsExpression = "#{@balanceParserProperties.getRetry().getMaxAttempts()}")
-    @Transactional
+    @Transactional(timeoutString = "#{@balanceParserProperties.getTransactionTimeout()}")
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/balance/AccountBalanceFileParser.java:79]: With my other suggestion:
```suggestion
    @Transactional(timeoutString = "#{@balanceParserProperties.getTransactionTimeout().toSeconds()}")
```

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/balance/AccountBalanceFileParser.java (hunk 2 above)
Comment (xin-hedera) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/balance/AccountBalanceFileParser.java]: updated

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/AbstractParserProperties.java (hunk 1 above)
Comment (xin-hedera) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/AbstractParserProperties.java]: updated

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/AbstractParserProperties.java [hunk 3]
```diff
@@ -50,6 +52,9 @@
     @NotNull
     protected RetryProperties retry = new RetryProperties();
 
+    @DurationMin(seconds = 30)
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/AbstractParserProperties.java:55]: Add `@NotNull`

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by Nana-EC — APPROVED** [review]
LGTM

**Review by ijungmann — APPROVED** [review]

**Pull request merged**